frontendActions/src/main/java/com/virtuslab/gitmachete/frontend/actions/toolbar/PushCurrentBranchAction.java

```diff
@@ -10,4 +16,20 @@
   public Option<String> getNameOfBranchUnderAction(AnActionEvent anActionEvent) {
     return getCurrentBranchNameIfManaged(anActionEvent);
   }
+
+  @Override
+  @UIEffect
+  public void onUpdate(AnActionEvent anActionEvent) {
+    var isEnabledAndVisible = getNameOfBranchUnderAction(anActionEvent)
+        .flatMap(bn -> getGitMacheteBranchByName(anActionEvent, bn))
+        .map(b -> b.getSyncToRemoteStatus().getRelation())
```

Comment:
I've also noticed a bit of inconsistency (maybe it isn't): the enum is called a `relation` in case of sync to remote but `status` in case of sync to parent... WDYT? maybe open an issue for v0.7.1

Reply:
issue https://github.com/VirtusLab/git-machete-intellij-plugin/issues/499 opened